### DISCUSSION TITLE
trigger search from url

### DISCUSSION THREAD

**Discussion by setop** [Pagefind/pagefind#733] [Ideas]
According to my research in issues and discussions, the topic has not been raised yet. Please correct me if I missed it.

I would find it useful to be able to share an url a website having pagefind search bar with the term to search in the URL, maybe as a fragment or a query string.

Like https://mdn.pagefind.app/#q=input or https://mdn.pagefind.app/?q=input will load page and immediately start searching.

![Screenshot from 2024-11-04 19-53-16](https://github.com/user-attachments/assets/6d655771-2d85-497b-8ef7-fc1edf34fca5)

**Comment by bglw** [comment]
👋 There's some prior art to look at for the Default UI — see https://github.com/CloudCannon/pagefind/issues/214 and https://kiko.io/post/Pagefind-UI-and-URL-Parameters/ for some discussion.

(At some point this is something Pagefind will look at officially handling, but that hasn't happened yet)

**Reply by rriemann** [reply]
This would be awesome. Any update since 2024?

**Reply by bglw** [reply]
No update as of yet, but I think it makes sense for this to come in the next release, so I'll do my best to fold it into the new component UI system in 1.6.0.

Added to the milestone in https://github.com/Pagefind/pagefind/issues/1164

**Comment by flyfoto** [comment]
I created my solution for this with the following JS. It also persists the current filter. However the latter part is not generic, since it knows which filters exist. Maybe someone knows how to improve it.
[searchbody.html](https://github.com/user-attachments/files/27788070/searchbody.html)
